File: Chapter1(Node_and_Topics)/indigo/scripts/minimal_publisher.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import Float64

rospy.init_node("minimal_publisher", anonymous = True)
pub = rospy.Publisher("topic1", Float64, queue_size = 10)

rate = rospy.Rate(10)

msg = Float64()
msg.data = 0.0
while not rospy.is_shutdown():
	pub.publish(msg)
	msg.data = msg.data + 0.01
	rate.sleep()

File: Chapter1(Node_and_Topics)/indigo/src/minimal_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/Float64.h>

int main(int argc, char **argv)
{
	/* code */
	ros::init(argc, argv, "minimal_publisher");
	ros::NodeHandle n;
	ros::Publisher pub = n.advertise<std_msgs::Float64>("topic1", 1);

	ros::Rate r(10);

	std_msgs::Float64 float_data;
	float_data.data = 0.0;

	while (ros::ok())
	{
		float_data.data = float_data.data + 0.001;
		pub.publish(float_data);
		r.sleep();
	}
	
}